person_storage.rb

```diff
@@ -0,0 +1,70 @@
+require_relative('./person')
+require_relative('./student')
+require_relative('./teacher')
+
+class PersonStorage
+  def initialize
+    @person_array = []
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_option = Integer(gets.chomp)
+
+    case person_option
+    when 1
+      print 'Age: '
+      user_age = Integer(gets.chomp)
+
+      print 'Name: '
+      user_name = gets.chomp
+
+      print 'Has parent permission? [Y/N]: '
+      user_permission = gets.chomp.to_s.upcase
+
+      case user_permission
+      when 'Y'
+        user_permission = true
+      when 'N'
+        user_permission = false
+      end
+      student = Student.new(user_age, user_name, parent_permission: user_permission)
+      @person_array.push({
+                           output: "[Student] Name: #{student.name}, ID: #{student.id}, Age: #{student.age}",
+                           object: student
+                         })
+
+      puts 'Person created successfully!'
+      puts "\n"
+
+    when 2
+      print 'Age: '
+      user_age = Integer(gets.chomp)
+
+      print 'Name: '
+      user_name = gets.chomp
+
+      print 'Specialization: '
+      user_specialization = gets.chomp
+
+      teacher = Teacher.new(user_age, user_name, user_specialization)
+      @person_array.push({
+                           output: "[Teacher] Name: #{teacher.name}, ID: #{teacher.id}, Age: #{teacher.age}",
+                           object: teacher
+                         })
+
+      puts 'Person created successfully!'
+      puts "\n"
+
+    else
+      puts 'Person not created.'
+    end
+  end
```

Comment:
- [ ] in this method you are still printing to the console, gathering information and processing data, which is not in line with the single responsibility rule as explained in the previous review. please split the method in more manageable pieces, below is an example of what this might look like:
```
  def create_person
    info = gather_person_info
    puts info
    if info["option"].to_i == 1 
      create_student(info)
    elsif info["option"].to_i == 2
      create_teacher(info)
    else
      return
    end
  end

  def gather_person_info
    new_person = {}
    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
    new_person["option"] = gets.chomp
    print 'Name: '
    new_person["name"] = gets.chomp
    print 'Age: '
    new_person["age"] = gets.chomp
    case new_person["option"]
    when '1'
      print 'Has parent permission? [Y/N]: '
      new_person["parent_permission"] = gets.chomp.downcase == 'y'
      puts 'Person created successfully!'
      return new_person
    when '2'
      print 'Specialization: '
      new_person["specialization"] = gets.chomp
      puts 'Person created successfully!'
      return new_person
    else
      puts 'That is not a valid input!'
      return
    end
  end

  def create_student(info)
    student = Student.new(name: info["name"], age: info["age"], parent_permission: info["parent_permission"], classroom: @classroom)
    puts student
    @people << student
  end

  def create_teacher(info)
    teacher = Teacher.new(name: info["name"], age: info["age"], specialization: info["specialization"])
    @people << teacher
  end
```